### ISSUE TITLE
Add a permission level where one can highlight but not add tags

### ISSUE THREAD

**Issue by remram44** [remram44/taguette#9]
This has been requested. It is useful when coding in a predetermined coding frame.

Is is easy to implement, however the level names should probably be revisited. Currently:

* Admin
* Can't change collaborators / delete project
* View and make changes
* (new level here: view and highlight?)
* View only

[See also the issue on GitLab: #153](https://gitlab.com/remram44/taguette/-/issues/153)

**Comment by Jash271** [comment]
Hey I'd like to look into this,could you assign this issue to me and also let me know in which endpoint do you want this functionality implemented ?

**Comment by remram44** [comment]
Hi @Jash271, welcome!

This is the enumeration that goes in the database: https://github.com/remram44/taguette/blob/14bf37413814a4ba681eb8f6d82281f3423f3695/taguette/database.py#L151. You can add a level there (`HIGHLIGHT_ONLY` is probably a good name for it?)

If you look for say `MANAGE_DOCS` you will find the places that deal with the privileges and you'll just have to add that one. There is just a couple of places in the frontend. The API itself shouldn't need changes, since it's just another value for the existing field.

Thanks!

**Comment by Jash271** [comment]
Okayy,so I only need to add this HIGHLIGHT_ONLY level in the privileges class ?

**Comment by remram44** [comment]
Yes and the places privileges are displayed. If you search for another privilege name you'll find them.

**Comment by Jash271** [comment]
Alright got it !

**Comment by Jash271** [comment]
So only the admin can add a tag right,while the other users can only highlight ?

**Comment by remram44** [comment]
No, there are 4 permission levels. The whole point is to add a level.

**Comment by Jash271** [comment]
Okayy👍

**Comment by remram44** [comment]
@Jash271 are you still planning to work on this?

**Comment by Jash271** [comment]
Yes I do plan to work on this,I have my exams going on currently so I am caught up with them at the moment ,once they are done,I'll make a pr asap
